### PR TITLE
Add x86-64-v2 & x86-64-v3 & v4 for Intel compilers

### PR DESCRIPTION
**Pull request by stephenmsachs**

### CODE REVIEW

File: cpu/microarchitectures.json [hunk 1]
```diff
@@ -145,6 +145,13 @@
             "flags": "-march={name} -mtune=generic -mcx16 -msahf -mpopcnt -msse3 -msse4.1 -msse4.2 -mssse3"
           }
         ],
+        "intel": [
+          {
+            "versions": "16.0:",
+            "name": "sse4.2",
+            "flags": "-m{name} -mtune=generic -mpopcnt"
```
Comment (alalazo) [inline cpu/microarchitectures.json:152]: Aren't we missing `-march=pentium4`? What would be the default if not specified?

**Comment by alalazo** [comment]
@rscohn2 Can you please double check these flags?

File: cpu/microarchitectures.json (hunk 1 above)
Comment (rscohn2) [reply cpu/microarchitectures.json:152]: icc defaults to SSE2

**Comment by rscohn2** [comment]
> @rscohn2 Can you please double check these flags?

I asked someone in the compiler team for a recommendation.

**Comment by rscohn2** [comment]
This is the recommendation for %intel (icc):
```
-march=corei7 for X86_64_v2
-march=core-avx2 for x86_64_v3
-march=core-avx512 for x86_64_v4
```

**Comment by alalazo** [comment]
Thanks @rscohn2 !

**Review by alalazo — APPROVED** [review]

**Pull request merged**